monai/transforms/intensity/array.py

```diff
@@ -297,6 +298,78 @@ def __call__(self, img: np.ndarray) -> np.ndarray:
         return scaler(img)
 
 
+class RandBiasField(Transform):
+    """
+    Random bias field augmentation for MR images.
+    The bias field is considered as a linear combination of smoothly varying basis (polynomial)
+    functions, as described in `Automated Model-Based Tissue Classification of MR Images of the Brain
+    <https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=811270>`_.
+    This implementation adapted from `NiftyNet
+    <https://github.com/NifTK/NiftyNet>`_.
+    Referred to `Longitudinal segmentation of age-related white matter hyperintensities
+    <https://www.sciencedirect.com/science/article/pii/S1361841517300257?via%3Dihub>`_.
+
+    Args:
+        degree: degree of freedom of the polynomials. Defaults to 3.
+        coeff_range: range of the random coefficients. Defaults to (0.0, 0.1).
+        dtype: output data type, defaut to float32.
+
+    """
+
+    def __init__(
+        self,
+        degree: int = 3,
+        coeff_range: Tuple[float, float] = (0.0, 0.1),
+        dtype: DtypeLike = np.float32,
+    ) -> None:
+        self.degree = degree
+        self.coeff_range = coeff_range
+        self.dtype = dtype
+
+    def _generate_random_field(
+        self,
+        image_shape: Tuple[int],
+        degree: int,
+        coeff_range: Tuple[float, float] = (0.0, 0.1),
+    ) -> np.ndarray:
+        """
+        products of polynomials as bias field estimations
+        """
+        rank = len(image_shape)
+        coeff_mat = np.zeros((degree + 1,) * rank)
+        n_coeff = int(np.prod([(degree + k) / k for k in range(1, rank + 1)]))
+        coeff = np.random.uniform(*coeff_range, n_coeff)
```

Comment:
I think you should inherrit from `Randomizable` and use local `self.R` instead of global random operations.